lms/lmsweb/views.py

```diff
@@ -83,27 +90,94 @@ def ratelimit_handler(e):
         return get_next_url(request.args.get('next'))
 
     username = request.form.get('username')
     password = request.form.get('password')
     next_page = request.form.get('next')
-    login_error = request.args.get('login_error')
+    login_message = request.args.get('login_message')
     user = User.get_or_none(username=username)
 
     if request.method == 'POST':
-        if user is not None and user.is_password_valid(password):
+        if (
+            user is not None and user.is_password_valid(password)
+            and not user.role.is_unverified
+        ):
             login_user(user)
             return get_next_url(next_page)
+
         elif user is None or not user.is_password_valid(password):
-            login_error = 'Invalid username or password'
-            error_details = {'next': next_page, 'login_error': login_error}
+            login_message = _('שם המשתמש או הסיסמה שהוזנו לא תקינים')
+            error_details = {'next': next_page, 'login_message': login_message}
+            return redirect(url_for('login', **error_details))
+
+        elif user.role.is_unverified:
+            login_message = _('עליך לאשר את המייל')
+            error_details = {'next': next_page, 'login_message': login_message}
             return redirect(url_for('login', **error_details))
 
-    return render_template('login.html', login_error=login_error)
+    return render_template('login.html', login_message=login_message)
+
+
+@webapp.route('/signup', methods=['GET', 'POST'])
+def signup():
+    form = RegisterForm()
+    if not form.validate_on_submit():
+        return render_template('signup.html', form=form)
+
+    user = User.get_or_create(**{
+        User.mail_address.name: form.email.data,
+        User.username.name: form.username.data,
+    }, defaults={
+        User.fullname.name: form.fullname.data,
+        User.role.name: Role.get_unverified_role(),
+        User.password.name: form.password.data,
+        User.api_key.name: User.random_password(),
+    })
+    send_confirmation_mail(user[0])
+    return redirect(url_for(
+        'login', login_message=_('ההרשמה בוצעה בהצלחה'),
+    ))
+
+
+@webapp.route('/confirm-email/<int:user_id>/<token>')
+def confirm_email(user_id: int, token: str):
+    try:
+        user = User.get_or_none(User.id == user_id)
+        if user is None:
+            return fail(404, f'No such user with id {user_id}.')
+
+        if not user.role.is_unverified:
+            return fail(
+                403, f'User has been already confirmed {user.username}',
+            )
+
+        SERIALIZER.loads(
+            token, salt=retrieve_salt(user), max_age=CONFIRMATION_TIME,
+        )
+        update = User.update(
+            role=Role.get_student_role(),
+        ).where(User.username == user.username)
+        update.execute()
+
+        return redirect(url_for(
+            'login', login_message=(
+                _('המשתמש שלך אומת בהצלחה, כעת הינך יכול להתחבר למערכת'),
```

Comment:
Changed it into `אתה` but now the whole sentence because he is already in the `login` page, the user just see the mesage that his user has been successfully confirmed and he can login